forge-game/src/main/java/forge/game/ability/effects/ChooseLetterEffect.java

```diff
@@ -0,0 +1,27 @@
+package forge.game.ability.effects;
+
+import com.google.common.collect.Lists;
+import forge.game.ability.AbilityUtils;
+import forge.game.ability.SpellAbilityEffect;
+import forge.game.card.Card;
+import forge.game.spellability.SpellAbility;
+
+import java.util.*;
+
+public class ChooseLetterEffect extends SpellAbilityEffect {
+
+    @Override
+    public void resolve(SpellAbility sa) {
+        final Card card = sa.getHostCard();
+        List<String> letters = Lists.newArrayList("A", "B", "C", "D", "E", "F", "G", "H", "I", "J", "K", "L", "M", "N",
+                "O", "P", "Q", "R", "S", "T", "U", "V", "W", "X", "Y", "Z");
+        List<String> consonants = Lists.newArrayList("B", "C", "D", "F", "G", "H", "J", "K", "L", "M", "N", "P", "Q",
+                "R", "S", "T", "V", "W", "X", "Y", "Z");
+        List<String> choices = sa.hasParam("Consonant") ? consonants : letters;
```

Comment:
Imo, I would have used Constants for this? And only make a new List if "Exclude" is used?

Also need extra list for Vowels

Reply:
Sounds good – I am not sure how to set that up so it would be good to learn

Reply:
Like this?

```java

List<String> LETTERS = Lists.newArrayList();
//... inside function:
List<String> choices = LETTERS;
if (sa.hasParam("Exclude"))  {
  choices = Lists.newArrayList(choices); // makes copy
  choices.removeAll(Arrays.asList(sa.getParam("Exclude").split(",")));
}


```

hm the `chooseLetter` implementations does remove the letter anyway from the list, so you can't give them the constant ? :(